### PR TITLE
Add archlinux integration tests for vulnerability detector

### PR DESCRIPTION
**Pull request by spothound**
|Related issue|
|---|
|Closes https://github.com/wazuh/wazuh-qa/issues/1119|

## Description

The following tests have been added for arch linux:

- Feed tests: These tests have to cover the addition of the new feed for Arch Linux
- Provider tests: Here we will test the settings for the new provider
-  Scan results tests: Here we have to add the tests that ensure Arch Linux agents are scanned properly
- [x] [Add Arch Linux to download feeds test](https://github.com/wazuh/wazuh-qa/commit/14dc3a3a89ef55afee8d47f64d9ac5057f7be6ce)
- [x] [Add Arch Linux to invalid type custom feed](https://github.com/wazuh/wazuh-qa/commit/8fab917ec96d2492213a02e17cb0df8d29801af9)
- [x] [Add Arch Linux to invalid type URL feeds](https://github.com/wazuh/wazuh-qa/commit/14b52d9bb694482bc085bb0990ac06ebc01880cc)
- [x] [Add Arch Linux to validate feed content](https://github.com/wazuh/wazuh-qa/commit/6e0000491846f005bb42716bbc1d9feaeae04eaf)
- [x] [Add Arch Linux's extra tag tests](https://github.com/wazuh/wazuh-qa/commit/dc070e64437430fbfbd5c50f68673fa19ce46d31)
- [x] [Add Arch Linux's invalid syntax feed test](https://github.com/wazuh/wazuh-qa/commit/a2435ce14e323bf171a7a86a6dfc71f9708a667c)
- [x] [Add Arch Linux's invalid values tests](https://github.com/wazuh/wazuh-qa/commit/bb2eabef9eb4126bdcb425a18abb4864f5eaf1b9)
- [x] [Add Arch Linux's missing fields test](https://github.com/wazuh/wazuh-qa/commit/5924975d5b46e734096fbaaa463966c0414eade5)

## Tests

- [x] Proven that tests **pass** when they have to pass.
- [x] Proven that tests **fail** when they have to fail.
- [x] Python codebase satisfies PEP-8 style style guide. `pycodestyle --max-line-length=120 --show-source --show-pep8 file.py`.
- [x] Python codebase is documented following the Google Style for Python docstrings.
- [x] The test is documented in wazuh-qa/docs.
- [x] `provision_documentation.sh` generate the docs without errors.

### CODE REVIEW

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture
+def modify_feed(test_values, request):
+    """
+    Modify the Arch Linux JSON feed, setting a test tag value
+    """
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py:45]:
```suggestion
    """Modify the Arch Linux JSON feed by setting a test tag value."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+    yield
+
+    write_json_file(custom_archlinux_json_feed_path, backup_data)
+    vd.clean_vuln_and_sys_programs_tables()
+    file.truncate_file(LOG_FILE_PATH)
+
+
+def test_no_feed_changes(clean_vuln_tables, get_configuration, configure_environment, restart_modulesd):
+    """Check if the feed is imported successfully by default"""
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py:75]:
```suggestion
    """Check if the feed is imported successfully by default."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    vd.check_feed_imported_successfully(wazuh_log_monitor=wazuh_log_monitor, log_system_name=vd.ARCH_LOG,
+                                        expected_vulnerabilities_number=vd.ARCH_NUM_CUSTOM_VULNERABILITIES)
+
+
+@pytest.mark.parametrize('test_values', vd.EXTRA_TEST_VALUES, ids=vd.EXTRA_TEST_IDS)
+def test_extra_tags_arch_linux_feed(test_values, clean_vuln_tables, get_configuration, configure_environment,
+                                    modify_feed):
+    """
+    Check if vulnerability detector behaves as expected when importing Arch Linux JSON feed with extra tags
+    """
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py:85]:
```suggestion
    """Check if Vulnerability Detector behaves as expected while importing Arch Linux JSON feed with extra tags."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture
+def modify_feed(test_data, request):
+    """
+    Modify the Arch Linux feed, setting a test field value
+    """
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py:86]:
```suggestion
    """Modify the Arch Linux feed by setting a test field value."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+    yield
+
+    write_json_file(custom_archlinux_json_feed_path, backup_data)
+    vd.clean_vuln_and_sys_programs_tables()
+    truncate_file(LOG_FILE_PATH)
+
+
+def test_no_feed_changes(clean_vuln_tables, get_configuration, configure_environment, restart_modulesd):
+    """Check if the feed is imported successfully by default"""
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py:111]:
```suggestion
    """Check if the feed is imported successfully by default."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    vd.check_feed_imported_successfully(wazuh_log_monitor=wazuh_log_monitor, log_system_name=vd.ARCH_LOG,
+                                        expected_vulnerabilities_number=vd.ARCH_NUM_CUSTOM_VULNERABILITIES)
+
+
+@pytest.mark.parametrize('test_data', test_data, ids=test_data_ids)
+def test_invalid_syntax_arch_linux_feed(test_data, clean_vuln_tables, get_configuration, configure_environment,
+                                        modify_feed):
+    """
+    Check if vulnerability detector behaves as expected when importing Arch Linux feed with syntax errors
+    """
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py:121]:
```suggestion
    """Check if Vulnerability Detector behaves as expected while importing Arch Linux feed with syntax errors."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture
+def modify_feed(test_data, custom_input, request):
+    """
+    Modify the Arch Linux feed, setting a test field value
+    """
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py:53]:
```suggestion
    """Modify the Arch Linux feed by setting a test field value."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+    yield
+
+    write_json_file(custom_archlinux_json_feed_path, backup_data)
+    vd.clean_vuln_and_sys_programs_tables()
+    truncate_file(LOG_FILE_PATH)
+
+
+def test_no_feed_changes(clean_vuln_tables, get_configuration, configure_environment, restart_modulesd):
+    """Check if the feed is imported successfully by default"""
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py:75]:
```suggestion
    """Check if the feed is imported successfully by default."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+                                        expected_vulnerabilities_number=vd.ARCH_NUM_CUSTOM_VULNERABILITIES)
+
+
+@pytest.mark.parametrize('test_data, custom_input', itertools.product(test_data, vd.CUSTOM_INPUT_TEST_VALUES),
+                         ids=test_data_ids)
+def test_invalid_values_arch_linux_feed(test_data, custom_input, clean_vuln_tables, get_configuration,
+                                        configure_environment, modify_feed):
+    """
+    Check if vulnerability detector behaves as expected when importing Arch Linux feed with wrong field values
+    """
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py:86]:
```suggestion
    """Check if Vulnerability Detector behaves as expected when importing Arch Linux feed with wrong field values."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope='module', params=fields, ids=field_ids)
+def remove_field_feed(request):
+    """
+    It allows to modify the feed by removing a certain field and loading the new feed configuration
+    """
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py:54]:
```suggestion
    """Modify the feed by removing a certain field and loading the new feed configuration."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+
+    yield request.param
+
+    write_json_file(custom_archlinux_json_feed_path, backup_data)
+    vd.clean_vuln_and_sys_programs_tables()
+    truncate_file(LOG_FILE_PATH)
+
+
+def test_no_feed_changes(clean_vuln_tables, get_configuration, configure_environment, restart_modulesd):
+    """Check if the feed is imported successfully by default"""
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py:74]:
```suggestion
    """Check if the feed is imported successfully by default."""
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py [hunk 12]
```diff
@@ -0,0 +1,94 @@
...
+    vd.check_feed_imported_successfully(wazuh_log_monitor=wazuh_log_monitor, log_system_name=vd.ARCH_LOG,
+                                        expected_vulnerabilities_number=vd.ARCH_NUM_CUSTOM_VULNERABILITIES)
+
+
+def test_invalid_archlinux_feed(clean_vuln_tables, get_configuration, configure_environment, remove_field_feed):
+    """
+    Check if vulnerability detector behaves as expected when importing Arch Linux feed with missing fields
+    """
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py:82]:
```suggestion
    """Check if Vulnerability Detector behaves as expected when importing Arch Linux feed with missing fields."""
```

**Review by BraulioV — CHANGES_REQUESTED** [review]
GJ Paco! I requested some minor changes in the PR related to the docstrings but anything else... LGTM. Just one more request: update the docs of the Vulnerability Detector tests.

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py (hunk 12 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py (hunk 11 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py:72]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py (hunk 10 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_missing_tags_archlinux_feed.py]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py (hunk 9 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py (hunk 8 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py:73]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py (hunk 7 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_values_archlinux_feed.py]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py (hunk 6 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py (hunk 5 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py:109]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py (hunk 4 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_invalid_syntax_archlinux_feed.py]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py (hunk 3 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py (hunk 2 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py (hunk 1 above)
Comment (spothound) [reply tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py]: https://github.com/wazuh/wazuh-qa/pull/1220/commits/7d650dcd2fb56e9cb3bd2810661d798e8a64e60b

File: docs/tests/integration/test_vulnerability_detector/test_feeds/test_invalid_type_url_feeds.md [hunk 13]
```diff
@@ -1,4 +1,64 @@
...
+platform linux -- Python 3.7.3, pytest-5.4.2, py-1.8.1, pluggy-0.13.1
+rootdir: /root/wazuh-qa/tests/integration, inifile: pytest.ini
+plugins: html-2.0.1, metadata-1.9.0, testinfra-5.0.0
+collected 40 items
+
+test_vulnerability_detector/test_feeds/test_invalid_type_url_feeds.py ....................... [ 57%]
+.......xxxxxxxxxx                                                                                [100%]
+
+============================== 30 passed, 10 xfailed in 83.46s (0:01:23) ===============================
+```
```
Comment (BraulioV) [inline docs/tests/integration/test_vulnerability_detector/test_feeds/test_invalid_type_url_feeds.md:60]:
```suggestion
```

File: tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py [hunk 14]
```diff
@@ -0,0 +1,90 @@
...
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+import json
+import os
+from copy import deepcopy
+
+import pytest
+import wazuh_testing.vulnerability_detector as vd
+from wazuh_testing.tools import LOG_FILE_PATH
+from wazuh_testing.tools import file
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_feeds/archlinux/test_extra_tags_archlinux_feed.py:11]:
```suggestion
from wazuh_testing.tools import LOG_FILE_PATH, file
```

File: tests/integration/test_vulnerability_detector/test_scan_results/test_archliux_inventory_archlinux_feed.py [hunk 15]
```diff
@@ -0,0 +1,83 @@
...
+    """
+    It allows to mock the vulnerability scan inserting custom packages, feeds and changing the host system
+    """
+    # Mock system
+    vd.modify_system(agent_id=mock_agent, os_name=os_name, os_major=os_major,
+                     os_minor=os_minor, name=vd.MOCKED_AGENT_NAME)
+
+    vulnerability = request.param
+
+    print(vulnerability)
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_scan_results/test_archliux_inventory_archlinux_feed.py:63]:
```suggestion
```

**Review by BraulioV — CHANGES_REQUESTED** [review]

**Review by BraulioV — APPROVED** [review]

**Pull request merged**